### DISCUSSION TITLE
Can you tell me if I'm making any mistakes with my setup?

### DISCUSSION THREAD

**Discussion by Alexconquer** [seaweedfs/seaweedfs#4727] [Q&A]
Hello,
I've just discovered SeaweedFS, and I find the software to be fantastic :D. Initially, I used GlusterFS, but I find SeaweedFS to be more user-friendly and flexible.

Anyway, I have a small infrastructure and I'd like your opinion to know if I'm making any mistakes:

Here's what I'm planning to do:

I have a server running ProxmoxVE:

VM storage with weedvolume
VM Manager with weedmaster + weedfiler mount with fuse) and Docker Swarm leader
2 VM Workers with weedfiler (mount with fuse) and Docker Swarm Worker 
Next to that, I have a Qnap NAS with a VM for a second weedvolume.
I think set master in 010 réplication.

I also plan to fix a server in the near future and add it to the ProxmoxVE cluster.

I intend to use SeaweedFS as network storage for all my containers.

I have Mysql and Sqlite databases, and I host my own Gitlab.

Is it good to store my databases on SeaweedFS?
Is it possible to use SeaweedFS at a lower level on ProxmoxVE and store VMs directly on SeaweedFS for HA (for example with VM manager )?

Thank you :D

**Comment by chrislusf** [comment]
I have not tried this setup before. Would like to know your result.

**Comment by Alexconquer** [comment]
Hello, I just got back from vacation.

Initially, I was looking for a flexible storage solution for Docker Swarm. I wanted to share storage between managers and workers, including setting files, docker-compose.yml, and container volumes.

In production, I use GlusterFS, but in smaller clusters, I experience poor file access performance. As a result, I have to use an external cluster for PostgreSQL.

I am looking for a new solution to replace GlusterFS, especially since its development has stopped.

For testing SeaweedFS, I used my home server, which consists of:

One server running Proxmox VE:
Manager VM
Worker VM
And a QNAP NAS with Container Station

The master and first volume are installed directly on Proxmox VE, using the SeaweedFS executable. The filer runs on the Manager VM, also using the SeaweedFS executable, and the second volume is on the QNAP NAS, running as a Docker image.

It took me a little time to understand the system, but ultimately, SeaweedFS is easy to use. Even in a small setup, SeaweedFS is very fast.

I tested it with multiple containers starting at the same time:

GitLab
VictoriaMetrics
Portainer
Teleport Agent
...and other small projects using SQLite
I experienced very good performance, even with GitLab, which is resource-intensive for smaller servers. The system is also robust against accidental data loss or disk saturation.

However, there is a significant "but."

With two PostgreSQL databases and two Prometheus instances in GitLab and VictoriaMetrics, there's a lot of file writing. That's when I understood the event log system. Over time, the event log folder became very large. For 13 GB of actual files, my volume increased to 90 GB in just one week, solely because of the event logs.

This was a big disappointment. I thought I had found the perfect solution for my servers, but I think it's still achievable.

My suggestions are:

Add a parameter in the filer to disable the event log. This would allow SeaweedFS to handle files with frequent writes. I understand that this would disable backup or sync, but I can manually backup using FUSE mount.
Another possibility, although more complex, would be to add a command that consolidates the event log folder and preserves only the latest write logs for all files.

I hope this helps! Let me know if you have any more questions.

**Reply by chrislusf** [reply]
You can manually delete the event logs for now.

**Reply by Alexconquer** [reply]
Certainly, for my server at home I can do it manually.
But not for servers in prod of work.
and do a cron to delete the log files... I don't find that elegant...

**Reply by chrislusf** [reply]
this needs some non-trivial eng work to compact the metadata change logs into snapshots, and be able to bootstrap from the snapshots when needed.

**Reply by Alexconquer** [reply]
is it possible to just add a "--disable-log-events" parameter in the filer command?
because, indeed, I think that this is easier than rewriting the metadata

**Comment by Alexconquer** [comment]
Finally, I have 2 Seaweed clusters:

- a "normal" cluster for backups and config files
- a high activity “HA” cluster

For the HA Cluster, I made a script to delete the meta data to avoid overload. i use filer API.

```
#!/bin/bash

# Variables
FILER_SERVER="http://xxx.xxx.xxx.xxx:9777"
DIR_PATH="/topics/.system/log"
TODAY=$(date +%Y-%m-%d)
YESTERDAY=$(date -d "yesterday" +%Y-%m-%d)

# Récupérer la liste des dossiers
DIRS=$(curl -s -H "Accept: application/json" "$FILER_SERVER$DIR_PATH/" | jq -r '.Entries[] | .FullPath | split("/")[-1]')

# Vérifier si le dossier avec la date d'aujourd'hui existe
if ! echo "$DIRS" | grep -q "$TODAY"; then
    # Si le dossier d'aujourd'hui n'existe pas, utiliser la date d'hier
    TODAY="$YESTERDAY"
fi

# Parcourir chaque dossier
for dir_name in $DIRS; do
    # Vérifier si le nom du dossier est une date valide
    if [[ "$dir_name" =~ ^[0-9]{4}-[0-9]{2}-[0-9]{2}$ && "$dir_name" < "$TODAY" ]]; then
        # Supprimer le dossier via l'API
        curl -X DELETE "$FILER_SERVER$DIR_PATH/$dir_name"
    fi
done
```

So I have my docker swarm cluster which is completely based on Seaweed, with active containers:

- Gitlab
- VictoriaMetrics
- MySql server
- and several sqllite db

Clearly, I have a significant gain in performance compared to GlusterFS... on a very small server system.

**Reply by ppenguin** [reply]
Even for a simple setup (2 volume servers, one filer/master, only CSI volumes in use via `nomad`) I got huge `/topics/.system/log` contents, amounting to 10TB over half a year for (now) 50GB of data payload. (It came as a surprise since I didn't bother to check after I got CSI volumes with swfs and `nomad` working.)

Is there a more sustainable (config?) solution yet, instead of brute force scripted deletion?

**Reply by chrislusf** [reply]
in `weed shell` there is a `fs.log.purge` command for this:

```
> fs.log.purge -h
Usage of fs.log.purge:
  -daysAgo uint
    	purge logs older than N days (default 365)
  -v	verbose mode
```

**Reply by ppenguin** [reply]
> in `weed shell` there is a `fs.log.purge` command for this:

Thanks! 
Is there (or could there be?) a config setting/flag that would purge logs older than X automatically (directly in the server?)
I realise this is a bit akin to what was suggested above, but this should only actually impact data state if more than one filer is used? And/or: why do we need the event log if we have only one filer? 
Or from the angle of transactions/journal concept: shouldn't logs be flushed when any pending syncs are complete?

**Comment by adhisimon** [comment]
Why not using logrotate to rotate and delete old logs by scheduled automatically?
